### PR TITLE
PT-3497: Rework scope picker to universal ui-scroll-drop-down

### PR DESCRIPTION
**Pull request by akak1977**
## Description
Scope picker reworked to an universal ui-scroll-drop-down that we use everywhere in order to have paged selector with the ability to search, multiple select, etc.
Before:
image
After:
image

### Jira-link:
https://virtocommerce.atlassian.net/browse/PT-3496

### CODE REVIEW

**Review by Vectorfield4 — APPROVED** [review]
image
Decrease padding-top between rows.

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/common/scope-value-pick-from-simple-list.js [hunk 1]
```diff
@@ -3,21 +3,27 @@ angular.module('platformWebApp')
...
-            blade.currentEntities = data;
-        }, function (error) {
-            bladeNavigationService.setError('Error ' + error.status, blade);
+        blade.selectedIds = _.map(blade.currentEntity.assignedScopes, x => x.scope);
+        $scope.items = _.map(blade.currentEntity.assignedScopes, function (x) {
+            var result = {};
+            result.id = x.scope;
+            result.name = x.label;
+            return result;
         });
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/common/scope-value-pick-from-simple-list.js:12]: how about restyling:

 $scope.items = _.map(blade.currentEntity.assignedScopes, x => {
            id: x.scope,
            name: x.label
        });

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/common/scope-value-pick-from-simple-list.js [hunk 2]
```diff
@@ -3,21 +3,27 @@ angular.module('platformWebApp')
...
-    }
+        blade.isLoading = false;
+    }       
+
+    $scope.search = function (criteria) {
+        return blade.dataService.search(criteria);
+    };
+
+    $scope.selectItem = function (item, id) {
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/common/scope-value-pick-from-simple-list.js:20]: remove redundant parameter

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/common/scope-value-pick-from-simple-list.js [hunk 3]
```diff
@@ -3,21 +3,27 @@ angular.module('platformWebApp')
...
+        $scope.items.push(item);
+    };
+
+    $scope.removeItem = function (item, id) {
```
Comment (eMazeika) [inline src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/common/scope-value-pick-from-simple-list.js:24]: remove redundant parameter

File: src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/common/scope-value-pick-from-simple-list.js (hunk 1 above)
Comment (akak1977) [reply src/VirtoCommerce.Platform.Web/wwwroot/js/app/security/blades/common/scope-value-pick-from-simple-list.js:12]: This does not work. 'Name' is a const.

**Review by ksavosteev — COMMENTED** [review]
I'd add a new blade, not edit the existing one

**Pull request merged**